### PR TITLE
rye-devtools: find-downloads packaging and improvements

### PR DESCRIPTION
**Pull request by dsp**
This PR is a propoosal to

1. move find-downloads to scripts
2. package scripts as a pyproject (only find-downloads atm)
3. move find-downloads to use asyncio

The idea here is to make it easier to manage dependencies for our scripts,
since i ran having to install httpx globally. It also makes it nicer to
dogfood rye on a regular basis.

If we are happy with this, i am okay moving the rest of the python scripts in scripts/
into the project.

Since I am not super familiar with writing python day to day, the whole setup might
be wrong or the choice of libraries. Happy to be corrected and iterate.
 
- scripts: make scripts a python project and move find-downloads there
- async fetching in find-downloads

```sh
$ time python3 find-downloads.py
    163.32 real         2.82 user         0.46 sys
$ time rye run find-downloads
    14.11 real         2.06 user         0.33 sys
```

This PR results in 11x speedup of find-downloads. Of course it's a simple,  one time measurement.

### CODE REVIEW

**Comment by dsp** [comment]
Notable, i am slowly overengineerng find-downloads ;)

File: pyproject.toml [hunk 1]
```diff
@@ -29,3 +29,4 @@ managed = true
 serve-docs = "mkdocs serve"
 
 [tool.rye.workspace]
+members = ["scripts"]
```
[the remark this answers is not in the document]
Comment (dsp) [reply pyproject.toml:32]: How about moving it to a directory called utilities?

[the remark this answers is not in the document]
Comment (T-256) [reply pyproject.toml:32]: I'd recommend move `find-downloads.py` to `scripts` and adds its dependencies to top-level `pyproject.toml`.

**Comment by T-256** [comment]
> Notable, i am slowly overengineerng find-downloads ;)

If you want, you can publish it as package so Rye can use it as dependency.

File: pyproject.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (mitsuhiko) [reply pyproject.toml:32]: I would probably make a python package called `rye-devtools` and then place the stuff there.

**Pull request merged**